client/tests/integration/events/notification.rs

```diff
@@ -34,13 +35,12 @@ fn trigger_completion_success_should_produce_event() -> Result<()> {
     let thread_client = test_client.clone();
     let (sender, receiver) = mpsc::channel();
     let _handle = thread::spawn(move || -> Result<()> {
-        let mut event_it = thread_client.listen_for_events(
-            NotificationEventFilter::TriggerCompleted(TriggerCompletedEventFilter::new(
-                Some(trigger_id),
-                Some(TriggerCompletedOutcomeType::Success),
-            ))
-            .into(),
-        )?;
+        let mut event_it =
+            thread_client.listen_for_events(NotificationEventFilter::ByTriggerCompleted(
+                TriggerCompletedEventFilter::new()
+                    .from_trigger(trigger_id)
+                    .with_outcome(TriggerCompletedOutcomeType::Success),
+            ))?;
```

Comment:
```suggestion
            thread_client.listen_for_events(
                TriggerCompletedEventFilter::new()
                    .from_trigger(trigger_id)
                    .with_outcome(TriggerCompletedOutcomeType::Success),
            )?;
```

I think this API should work you just have to implement `From<TriggerCompletedEventFilter> for `EventFilterBox`

Reply:
Actually, why does `NotificationEvent` have a single variant? Maybe we should flatten it like the `ExecuteTrigger`?

@Erigara I see that you added the `NotificationEvent` in #3789. Did you have any plans/ideas to add more kinds of notification events?

Reply:
Discussed with Roman in PM, he's fine with removing it. `NotificationEvent` was just supposed to be an event that does not trigger triggers, with no concrete plans for more events in that category.

Reply:
Removed now, conversion is done implicitly now